scl/graylog2/plugin.conf

```diff
@@ -23,7 +23,7 @@
 
 @requires json-plugin
 
-template-function "format-gelf" "$(format-json version='1.1' host='${HOST}' short_message='${MSG}' level=int(${LEVEL_NUM}) timestamp=int64(${R_UNIXTIME}) _program='${PROGRAM}' _pid=int(${PID}) _facility='${FACILITY}' _class='${.classifier.class}' --key .* --key _*)$(binary 0x00)";
+template-function "format-gelf" "$(format-json --omit-empty-values version='1.1' host='${HOST}' short_message='${MSG}' level=int(${LEVEL_NUM}) timestamp=int64(${R_UNIXTIME}) _program='${PROGRAM}' _pid=int(${PID}) _facility='${FACILITY}' _class='${.classifier.class}' --key .* --key _*)$(binary 0x00)";
```

Comment:
As you mentioned, adding `--omit-empty-values ` is not enough, because `GELF` specifies mandatory fields:
version, host, and short_message.

In rare cases, $MSG can be empty, and we should not generate invalid output, so I'd also suggest adding a default value to `short_message`.